### PR TITLE
add initial sv filtering tool

### PR DESCRIPTION
**Pull request by apaul7**
This PR adds the cwl tool for running bcftools to filter Manta and Snoove SVs based on issue #715 . 

Future PR will add this and AnnotSV to single sample sv caller subworkflow

### CODE REVIEW

File: definitions/tools/filter_sv_vcf.cwl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        default: 0.2
+        inputBinding:
+            position: 4
+        doc: "required alternate paired read abundance percentage to pass"
+    split_count:
+        type: int?
+        default: 2
+        inputBinding:
+            position: 5
+        doc: " if present in variant, number of alternate split reads support needed to pass"
```
Comment (tmooney) [inline definitions/tools/filter_sv_vcf.cwl]:
```suggestion
        doc: "if present in variant, number of alternate split reads support needed to pass"
```

File: definitions/tools/filter_sv_vcf.cwl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          split_count="$5"
+          split_perc="$6"
+          vcf_source="$7"
+
+          if [ "$vcf_source" == "smoove" ]; then
+            echo "Running filter for smoove vcf"
+            filter_expression="(AS >= $split_count && (AS / (RS+AS) >= $split_perc)) && (AP >= $paired_count && (AP / (AP+RP) >= $paired_perc))"
+          elif [ "$vcf_source" ==  "manta" ]; then
+            echo "Running filter for manta vcf"
+            filter_expression="(SR[0:*]=\".\" || (SR[0:1] >= 2 && (SR[0:1] / (SR[0:0]+SR[0:1]) >= $split_perc))) && (PR[0:1] >= $paired_count && (PR[0:1] / (PR[0:0]+PR[0:1]) >= $paired_perc))"
```
Comment (tmooney) [inline definitions/tools/filter_sv_vcf.cwl]: It looks like `$split_count` isn't used in the manta version.  I'd guess it goes here?
```suggestion
            filter_expression="(SR[0:*]=\".\" || (SR[0:1] >= $split_count && (SR[0:1] / (SR[0:0]+SR[0:1]) >= $split_perc))) && (PR[0:1] >= $paired_count && (PR[0:1] / (PR[0:0]+PR[0:1]) >= $paired_perc))"
```

File: definitions/tools/filter_sv_vcf.cwl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+requirements:
+    - class: ResourceRequirement
+      ramMin: 4000
+    - class: DockerRequirement
+      dockerPull: "mgibio/bcftools-cwl:1.9"
+    - class: InitialWorkDirRequirement
+      listing:
+      - entryname: "filter_sv_vcf_read_support.sh"
+        entry: |
+          set -eou pipefail
```
Comment (tmooney) [inline definitions/tools/filter_sv_vcf.cwl]: Not strictly necessary, but might be nice to have a `#!/bin/bash` line atop this file.

Comment (apaul7) [reply definitions/tools/filter_sv_vcf.cwl]: done!

**Comment by chrisamiller** [comment]
How were these default cutoffs decided upon?  I don't think they're necessarily bad, but it does seem like you're requiring both split and paired read support, which is fairly stringent, IIRC.

**Comment by apaul7** [comment]
Dave suggested requiring 2 paired/split reads and the 20% abundance. The split reads are only required if there is a field for them in the call. 
With these filtering parameters I've been seeing ~90% of the calls being filtered out and ~10% passing.

**Pull request merged**